onadata/libs/serializers/organization_member_serializer.py

```diff
@@ -161,25 +116,17 @@ def create(self, validated_data):
         if username:
             user = User.objects.get(username=username)
 
-            add_user_to_organization(organization, user)
-            _set_organization_role_to_user(organization, user, role)
-
-            if email_msg:
-                _compose_send_email(organization, user, email_msg, email_subject)
-
             if remove:
-                remove_user_from_organization(organization, user)
-
-            projects = Project.objects.filter(
-                organization=organization.user, deleted_at__isnull=True
-            )
-            for project in projects.iterator():
-                # Queue permission propagation with a
-                # delay for permissions to be effected
-                propagate_project_permissions_async.apply_async(
-                    args=[project.id], countdown=60
+                remove_org_user_async.apply_async(args=[organization.pk, user.pk])
+
+            else:
+                add_org_user_and_share_projects_async.apply_async(
+                    args=[organization.pk, user.pk, role]
                 )
 
+                if email_msg:
+                    _compose_send_email(organization, user, email_msg, email_subject)
+
```

Comment:
Should this email be sent after the permissions are successfully applied? As such, should it be triggered by `add_org_user_and_share_projects_async`?